slurmdb-fix-job-null-accounts.sh

```diff
@@ -2,17 +2,12 @@
 
 while read -r line
 do
-
 	uid=`echo $line | cut -d ',' -f1`
-	slurm_acct_name=`echo $line | cut -d ',' -f2`
-
-	total_jobs=`mysql -N -u root -D slurmdb -e "select count(1) from ohpc_job_table where id_user=$uid and account IS NULL;"`
-
-	#printf "User $slurm_acct_name with user ID $uid has $total_jobs jobs with account name null \n"
-
-	#printf "Updating $total_jobs records for user $slurm_acct_name with user ID $uid \n"
-
-	printf "UPDATE ohpc_job_table SET account=\'$slurm_acct_name\' WHERE id_user=$uid and account IS NULL;\n"
-
-	#mysql -u root -D slurmdb -e \"UPDATE ohpc_job_table SET account=$slurm_acct_name WHERE id_user=$uid and account IS NULL;\"
+	slurm_user_name=`echo $line | cut -d ',' -f2`
+	data1=$(mysql -u root -D slurmdb<<<"SELECT distinct(id_assoc) , acct, user  FROM ohpc_assoc_table  where user='$slurm_user_name';")
+	id_assoc=`echo $data1 | cut -d ' ' -f4`
+	acct=`echo $data1 | cut -d ' ' -f5`
+	user=`echo $data1 | cut -d ' ' -f6`
+	data1=$(mysql -u root -D slurmdb<<<"UPDATE ohpc_job_table  set id_assoc='$id_assoc' where id_user='$uid' and id_assoc=0;\
```

Comment:
Please print the update statements you intend to apply like I did on line 15 in the org file(you may have two update statements), it's a good practice to do it this way because you can have the exact number of update statements that you can match to the number of problematic records you observed from your select statement. 
While applying this patch in the prod you can write the update stmts to a file then apply all of them at once, this can improve confidence by giving us a chance to do sanity checks before hand rather than after the fact.

Reply:
While applying this patch in the prod you can write the update stmts to a file then apply all of them at once, this can improve confidence by giving us a chance to do sanity checks before hand rather than after the fact.